LibertyDSNP/spec
# enable tag based trigger

Problem
=======
## Acceptance Criteria
- [x] Update deploy workflow to use mdbook build
- [x] Should be dependent on CI lint passing
- [x] Should only deploy when tagged with a date based deployment regex `^\d{8}\.\d+`
- [x] Update readme to detail the date based deployment versioning.
Link to GitHub Issue(s): https://github.com/LibertyDSNP/spec/issues/157

Solution
========
- [x] ensure the cicd workflow for mdbook triggers on tag creation
- [x] step one should only trigger when date based tag is created via regex checks
- [x] deployment step should not trigger for PRs

Change summary:
---------------
- [x] Updated mdbook_ci.yaml to include three conditions above
- [x] Updated ReadMe with instructions to deploy via date tag

Steps to Verify:
----------------
1. PR checks should not run deployment step 
2.  Create  a temporary tag without following regex ^\d{8}\.\d+, build should not trigger in this case

Test:
- [x] Ran tests to ensure deployment workflow is picking up the correct sha of last commit on a branch
